python/lsst/meas/algorithms/loadIndexedReferenceObjects.py

```diff
@@ -136,3 +140,50 @@ def _trim_to_circle(self, catalog_shard, ctrCoord, radius):
             if record.getCoord().separation(ctrCoord) < radius:
                 temp_cat.append(record)
         return temp_cat
+
+
+class LoadPanstarrsObjectsTask(LoadIndexedReferenceObjectsTask):
+    """Reference catalog loader for ps1_pv3_3pi_20170110 catalog
+
+    This catalog is in the "LSST format" suitable for use with
+    the `LoadIndexedReferenceObjectsTask`, but has a slightly
+    different format for the proper motion columns than was
+    eventually adopted for the default of the LSST format:
+
+    - The proper motions are in units of arcsec/yr instead of mas/yr.
+    - The epoch is in seconds since the Unix epoch instead of MJD.
+
+    Overriding the ``calculateProperMotionScales`` method is
+    sufficient to remedy these.
+    """
+    def calculateProperMotionScales(self, catalog, epoch):
+        """Calculate the scaling factor(s) for proper motion
+
+        The scaling factor is the number (or numbers, if the value
+        varies per source) by which to multiply the proper motion
+        rates in the catalog in order to get the proper motion.
+        It means calculating the amount of time between the
+        reference catalog epoch and the desired epoch, and specifying
+        the units of proper motion.
+
+        Parameters
+        ----------
+        catalog : `lsst.afw.table.SimpleCatalog`
+            Reference catalog. By default, this contains a column
+            ``epoch``, which is the mean epoch of the source as a
+            Modified Julian Date (MJD) in TAI.
+        epoch : `float`
+            Epoch to which to move objects (TAI MJD).
+
+        Returns
+        -------
+        struct : `lsst.pipe.base.Struct`
+            Struct of proper motion scales, containing:
+
+            - ``values`` : values of the proper motion scale (may be
+                either a scalar `float` or an `numpy.ndarray`)
+            - ``units`` : units of proper motion (`lsst.afw.geom.Angle`)
+        """
+        epochUnix = DateTime(float(epoch), DateTime.TAI).nsecs()/1.0e9  # Sec since Unix epoch
+        timeDiff = (epochUnix - catalog["epoch"])/(3600*24*365.25)  # Time difference, years
```

Comment:
This feels like an good use case for `astropy.time`. Or do we have an appropriate time model in DateTime? `365.25` isn't technically precise, is it?

Reply:
3600 sec/hour *24 hours/day * 365.25 days is indeed defined to be what astronomers mean when they say "year" (where the sec is an SI second).  And thus if an astronomer quotes to you a proper motion/year, this is the year they were supposed to have used.

https://www.iau.org/publications/proceedings_rules/units/

Reply:
Is that going to be as fast?

Reply:
I think it will be fast enough. The time difference per object is an array operation (which is no guarantee, but it is bound to help).